src/main/java/com/elastic/support/diagnostics/commands/RunKibanaQueries.java

```diff
@@ -0,0 +1,309 @@
+    * Return will be used as workaround to Unit test. Will be replaced in v9
+    *
+    * @param  DiagnosticContext context
+    * @return SystemCommand
+    */
+    public SystemCommand execSystemCommands(DiagnosticContext context) {
+
+        ProcessProfile nodeProfile = getNodeProfile(context.tempDir, "kibana_node_stats.json", context);
+
+        if (StringUtils.isEmpty(nodeProfile.pid) || nodeProfile.pid.equals("1")) {
+            context.dockerPresent = true;
+            context.runSystemCalls = false;
+        }
+        // Create and cache the system command type we need, local or remote...
+        SystemCommand syscmd = null;
+        switch (context.diagnosticInputs.diagType) {
+            case Constants.kibanaRemote:
+                String targetOS;
+                if(context.dockerPresent){
+                    targetOS = Constants.linuxPlatform;
+                }
+                else{
+                    targetOS = nodeProfile.os;
+                }
+                
+                syscmd = getRemoteSystem(targetOS, context);
+                break;
+
+            case Constants.kibanaLocal:
+                if (context.dockerPresent) {
+                    syscmd = getLocalSystem("docker", true);
+                } else {
+                    syscmd = getLocalSystem(nodeProfile.os, false);
+                }
+                break;
+        }
+        return syscmd;
+    }
+
+
+    /**
+    * We will not collect all the headerst hat are returned by the actions API (kibana_actions.json file).
+    * for troubleshooting support engineers will only need "kbn-xsrf" or "Content-Type", all the others are removed.
+    *
+    * @param  DiagnosticContext context
+    * @return void
+    */
+    public void allowedHeadersFilter(DiagnosticContext context) {
+        JsonNode actions = JsonYamlUtils.createJsonNodeFromFileName(context.tempDir, "kibana_actions.json");
+        Boolean headerRemoved = false;
+        if (actions.size() > 0) {
+            for (int i = 0; i < actions.size(); i++) {
+                JsonNode config = actions.get(i).get("config");
+                // API webhook format can change, and maybe we have a webhook without config
+                if (!(config == null || config.isNull())) {
+                    // Not all the webhook need to have headers, so we need to be sure the data was set by the customer.
+                    JsonNode headers = actions.get(i).get("config").get("headers");
+                    if (!(headers == null || headers.isNull())) {
+                        Iterator<Map.Entry<String, JsonNode>> iter = actions.get(i).get("config").get("headers").fields();
+
+                        while (iter.hasNext()) {
+                            Map.Entry<String, JsonNode> entry = iter.next();
+                            if (!entry.getKey().equals("kbn-xsrf") && !entry.getKey().equals("Content-Type")) {
```

Comment:
It's worth noting that HTTP header keys are technically case insensitive. Kibana is almost certainly doing this consistently, but it's probably better to get the lowercased version of the key.

```suggestion
                            String key = entry.getKey().toLowerCase();

                            if (!key.equals("kbn-xsrf") && !key.equals("content-type")) {
```